src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -371,105 +375,49 @@ public MethodInfo(AssemblyInfo assembly, MethodDefinitionHandle methodDefHandle,
-                                break;
-                            case "DebuggerNonUserCodeAttribute":
-                                DebuggerAttrInfo.HasNonUserCode = true;
-                                break;
-                            case "DebuggerStepperBoundaryAttribute":
-                                DebuggerAttrInfo.HasStepperBoundary = true;
-                                break;
-                            case nameof(CompilerGeneratedAttribute):
-                                IsCompilerGenerated = true;
-                                break;
-                        }
-
-                    }
-                }
-                DebuggerAttrInfo.ClearInsignificantAttrFlags();
-            }
-            if (pdbMetadataReader != null)
-            {
-                localScopes = pdbMetadataReader.GetLocalScopes(methodDefHandle);
-                byte[] scopeDebugInformation =
-                        (from cdiHandle in pdbMetadataReader.GetCustomDebugInformation(methodDefHandle)
-                        let cdi = pdbMetadataReader.GetCustomDebugInformation(cdiHandle)
-                        where pdbMetadataReader.GetGuid(cdi.Kind) == PortableCustomDebugInfoKinds.StateMachineHoistedLocalScopes
-                        select pdbMetadataReader.GetBlobBytes(cdi.Value)).FirstOrDefault();
-
-                if (scopeDebugInformation != null)
-                {
-                    _asyncScopes = new AsyncScopeDebugInformation[scopeDebugInformation.Length / 8];
-                    for (int i = 0; i < _asyncScopes.Length; i++)
-                    {
-                        int scopeOffset = BitConverter.ToInt32(scopeDebugInformation, i * 8);
-                        int scopeLen = BitConverter.ToInt32(scopeDebugInformation, (i * 8) + 4);
-                        _asyncScopes[i] = new AsyncScopeDebugInformation(scopeOffset, scopeOffset + scopeLen);
+                        case "DebuggerHiddenAttribute":
+                            DebuggerAttrInfo.HasDebuggerHidden = true;
+                            break;
+                        case "DebuggerStepThroughAttribute":
+                            DebuggerAttrInfo.HasStepThrough = true;
+                            break;
+                        case "DebuggerNonUserCodeAttribute":
+                            DebuggerAttrInfo.HasNonUserCode = true;
+                            break;
+                        case "DebuggerStepperBoundaryAttribute":
+                            DebuggerAttrInfo.HasStepperBoundary = true;
+                            break;
+                        case nameof(CompilerGeneratedAttribute):
+                            IsCompilerGenerated = true;
+                            break;
                     }
                 }
-
-                _asyncScopes ??= Array.Empty<AsyncScopeDebugInformation>();
             }
+            if (!hasDebugInformation)
+                DebuggerAttrInfo.HasNonUserCode = true;
```

Comment:
Its used to decide if should skip method while stepping, I mean, if we don't have debug symbols (while loading the app), it means this library HasNonUserCode, otherwise we should load the pdb. Then when we are debugging and stepping we use this flag together with JustMyCode setting to decide if we should skip a pause in a non user code.